[Bug][Sort] NPE occurs when running Oracle all database migration jobs

### What happened

The following error occurs when enabling incremental snapshots during all database migration job.

```css
Caused by: java.lang.NullPointerException
	at org.apache.inlong.sort.cdc.oracle.debezium.table.RowDataDebeziumDeserializeSchema$18.convert(RowDataDebeziumDeserializeSchema.java:723) ~[pool-1-thread-8-1679744496795-dev-sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.debezium.table.RowDataDebeziumDeserializeSchema$13.convert(RowDataDebeziumDeserializeSchema.java:506) ~[pool-1-thread-8-1679744496795-dev-sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.debezium.table.RowDataDebeziumDeserializeSchema.extractAfterRow(RowDataDebeziumDeserializeSchema.java:827) ~[pool-1-thread-8-1679744496795-dev-sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.debezium.table.RowDataDebeziumDeserializeSchema.deserialize(RowDataDebeziumDeserializeSchema.java:799) ~[pool-1-thread-8-1679744496795-dev-sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.debezium.table.RowDataDebeziumDeserializeSchema.deserialize(RowDataDebeziumDeserializeSchema.java:788) ~[pool-1-thread-8-1679744496795-dev-sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.source.reader.OracleRecordEmitter.emitElement(OracleRecordEmitter.java:96) ~[pool-1-thread-8-1679744496795-dev-sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.source.reader.OracleRecordEmitter.processElement(OracleRecordEmitter.java:87) ~[pool-1-thread-8-1679744496795-dev-sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
	at org.apache.inlong.sort.cdc.oracle.shaded.org.apache.inlong.sort.cdc.base.source.reader.IncrementalSourceRecordEmitter.emitRecord(IncrementalSourceRecordEmitter.java:87) ~[pool-1-thread-8-1679744496795-dev-sort-connector-oracle-cdc-1.5.0-SNAPSHOT.jar:2.3.0]
```


### What you expected to happen

Run job normally.


### How to reproduce

Run all database migration job.

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/source/meta/split/SourceSplitState.java
+++ b/inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/source/meta/split/SourceSplitState.java
@@ -53,7 +53,7 @@ public abstract class SourceSplitState {
     /** Use the current split state to create a new SourceSplit. */
     public abstract SourceSplitBase toSourceSplit();
 
-    /** Get the current MySQLSplit. */
+    /** Get the current split. */
     @Internal
     public SourceSplitBase getSourceSplitBase() {
         return split;

--- a/inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/source/meta/split/SourceSplitState.java
+++ b/inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/source/meta/split/SourceSplitState.java
@@ -17,6 +17,8 @@
 
 package org.apache.inlong.sort.cdc.base.source.meta.split;
 
+import org.apache.flink.annotation.Internal;
+
 /** State of the reader, essentially a mutable version of the {@link SourceSplitBase}.
  * Copy from com.ververica:flink-cdc-base:2.3.0.
  * */
@@ -50,4 +52,10 @@ public abstract class SourceSplitState {
 
     /** Use the current split state to create a new SourceSplit. */
     public abstract SourceSplitBase toSourceSplit();
+
+    /** Get the current MySQLSplit. */
+    @Internal
+    public SourceSplitBase getSourceSplitBase() {
+        return split;
+    }
 }

--- a/inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/source/reader/OracleRecordEmitter.java
+++ b/inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/source/reader/OracleRecordEmitter.java
@@ -27,8 +27,11 @@ import com.ververica.cdc.debezium.history.FlinkJsonTableChangeSerializer;
 import io.debezium.connector.AbstractSourceInfo;
 import io.debezium.data.Envelope;
 import io.debezium.document.Array;
+import io.debezium.relational.TableId;
 import io.debezium.relational.history.HistoryRecord;
 import io.debezium.relational.history.TableChanges;
+import io.debezium.relational.history.TableChanges.TableChange;
+import java.util.Map;
 import org.apache.flink.api.connector.source.SourceOutput;
 import org.apache.flink.util.Collector;
 import org.apache.inlong.sort.cdc.base.debezium.DebeziumDeserializationSchema;
@@ -37,6 +40,7 @@ import org.apache.inlong.sort.cdc.base.source.meta.offset.OffsetFactory;
 import org.apache.inlong.sort.cdc.base.source.meta.split.SourceSplitState;
 import org.apache.inlong.sort.cdc.base.source.metrics.SourceReaderMetrics;
 import org.apache.inlong.sort.cdc.base.source.reader.IncrementalSourceRecordEmitter;
+import org.apache.inlong.sort.cdc.base.util.RecordUtils;
 import org.apache.kafka.connect.data.Struct;
 import org.apache.kafka.connect.source.SourceRecord;
 import org.slf4j.Logger;
@@ -76,7 +80,7 @@ public class OracleRecordEmitter<T> extends IncrementalSourceRecordEmitter<T> {
                 splitState.asStreamSplitState().recordSchema(tableChange.getId(), tableChange);
             }
             if (includeSchemaChanges) {
-                emitElement(element, output, splitState);
+                emitElement(element, output);
             }
         } else if (isDataChangeRecord(element)) {
             if (splitState.isStreamSplitState()) {
@@ -84,33 +88,33 @@ public class OracleRecordEmitter<T> extends IncrementalSourceRecordEmitter<T> {
                 splitState.asStreamSplitState().setStartingOffset(position);
             }
             reportMetrics(element);
-            emitElement(element, output, splitState);
+            final Map<TableId, TableChange> tableSchemas =
+                    splitState.getSourceSplitBase().getTableSchemas();
+            final TableChange tableSchema =
+                    tableSchemas.getOrDefault(RecordUtils.getTableId(element), null);
+            debeziumDeserializationSchema.deserialize(element, new Collector<T>() {
+
+                @Override
+                public void collect(T record) {
+                    Struct value = (Struct) element.value();
+                    Struct source = value.getStruct(Envelope.FieldName.SOURCE);
+                    String dbName = source.getString(AbstractSourceInfo.DATABASE_NAME_KEY);
+                    String schemaName = source.getString(AbstractSourceInfo.SCHEMA_NAME_KEY);
+                    String tableName = source.getString(AbstractSourceInfo.TABLE_NAME_KEY);
+                    sourceReaderMetrics
+                            .outputMetrics(dbName, schemaName, tableName, splitState.isSnapshotSplitState(), value);
+                    output.collect(record);
+                }
+
+                @Override
+                public void close() {
+
+                }
+            }, tableSchema);
         } else {
             // unknown element
             LOG.info("Meet unknown element {}, just skip.", element);
         }
     }
 
-    protected void emitElement(SourceRecord element, SourceOutput<T> output,
-            SourceSplitState splitState) throws Exception {
-        debeziumDeserializationSchema.deserialize(element, new Collector<T>() {
-
-            @Override
-            public void collect(T record) {
-                Struct value = (Struct) element.value();
-                Struct source = value.getStruct(Envelope.FieldName.SOURCE);
-                String dbName = source.getString(AbstractSourceInfo.DATABASE_NAME_KEY);
-                String schemaName = source.getString(AbstractSourceInfo.SCHEMA_NAME_KEY);
-                String tableName = source.getString(AbstractSourceInfo.TABLE_NAME_KEY);
-                sourceReaderMetrics
-                        .outputMetrics(dbName, schemaName, tableName, splitState.isSnapshotSplitState(), value);
-                output.collect(record);
-            }
-
-            @Override
-            public void close() {
-
-            }
-        });
-    }
 }

